beets/autotag/__init__.py

```diff
@@ -80,10 +85,12 @@ def apply_item_metadata(item: Item, track_info: TrackInfo):
     item.artist_sort = track_info.artist_sort
     item.artist_credit = track_info.artist_credit
     item.title = track_info.title
-    item.mb_trackid = track_info.track_id
-    item.mb_releasetrackid = track_info.release_track_id
-    if track_info.artist_id:
-        item.mb_artistid = track_info.artist_id
+    fields = [('mb_trackid', track_info.track_id),
+              ('mb_releasetrackid', track_info.release_track_id),
+              ('mb_artistid', track_info.artist_id)]
+    for field, value in fields:
+        if re.match(MBID_REGEX, str(value)) or item.get(field) is None:
+            item[field] = value
```

Comment:
Would it suffice to just check whether these fields are None? That is, something like this:

```py
for field in ['mb_trackid', 'mb_releasetrackid', 'mb_artistid']:
    value = track_info.get(field)
    if value:
         item[field] = value
```

Reply:
Actually, that may not be enough as it will prevent overwriting the `mb_*` fields (should the user choose to update tags using another MB release. The current approach allows updating the `mb_*` fields if it is empty or if the new value is valid. Hope that makes sense.

Reply:
I guess the question I was getting at was: do we really need to check for validity? It makes the logic here quite a bit more complex, and it would be really nice to avoid the need for the special case—hopefully by just not supplying invalid data in the first place.